### DISCUSSION TITLE
Finetuning without pre-tokenizing

### DISCUSSION THREAD

**Discussion by hanoonaR** [axolotl-ai-cloud/axolotl#1229] [Q&A] [answered]
Hi @winglian,

I have a question regarding fine-tuning a model without pre-tokenizing. I am unclear about the correct configuration settings in this context.

If the original fine-tuning with pre-tokenize for the dataset is as follows:
```
datasets:
    path: mhenrichsen/alpaca_2k_test
    type: alpaca
```

Would replacing it with the following be appropriate for this need:
```
pretraining_dataset:
  - path: mhenrichsen/alpaca_2k_test
    type: alpaca
```
    
Would replacing the initial configuration with the `pretraining_dataset` configuration be suitable for my purpose of fine-tuning without pre-tokenizing? Are there any specific implications or differences I should be aware of when opting for the `pretraining_dataset` configuration over the datasets configuration in this context?

I look forward to your guidance, Thank you in advance.

**Comment by NanoCode012** [comment]
Could you clarify without pre-tokenizing? The dataset in your first code block is not tokenized. Axolotl tokenizes it.

**Comment by NanoCode012** [comment, marked as answer]
Super late update. On re-reading this, `datasets` does pre-tokenization. `pretraining_datasets` use streaming. Differences are now properly documented here https://docs.axolotl.ai/docs/dataset-formats/
